File: solutions/elixir/05-iz3/code/lib/main.ex
defmodule CLI do
  def main(_args) do
    loop()
  end

  defp loop do
    IO.write("$ ")
    command = IO.gets("") |> String.trim()

    case command do
      "exit" -> :ok
      "echo " <> args ->
        IO.puts(args)
        loop()
      _ ->
        IO.puts("#{command}: command not found")
        loop()
    end
  end
end

File: solutions/ruby/05-iz3/code/app/main.rb
loop do
  $stdout.write("$ ")
  $stdout.flush
  command = $stdin.gets
  break if command.nil?

  command = command.chomp
  if command == "exit"
    break
  elsif command == "echo" || command.start_with?("echo ")
    $stdout.puts(command[5..] || "")
  else
    $stdout.puts("#{command}: command not found")
  end
end
